### PR TITLE
Handle upgrade option api errors so as to not spam sentry

### PR DESCRIPTION
**Pull request by tavvy**

### CODE REVIEW

File: server/controllers/subscription-transitions/get-upgrade-options-insecure.js [hunk 1]
```diff
@@ -13,7 +14,11 @@ export default function (req, res) {
 		.then(getUserUuid, {as: 'userUuid'})
 		.then(getUserProfile, {as: 'userProfileResult'})
 		.then(getUpgradeOptions, {as: 'apiResult'})
-		.then(setupThe(ApiResult))
+		.then(
+			ifThe(offerOrUserHasNotBeenFound)
+				(respondWith(getApiErrorResponse))
+		)
+		.otherwise(setupThe(ApiResult))
```
Comment (geek-caroline) [inline server/controllers/subscription-transitions/get-upgrade-options-insecure.js]: ooh, I like this promise handling flow, not seen this before

File: server/controllers/subscription-transitions/get-upgrade-options-insecure.js [hunk 2]
```diff
@@ -53,7 +58,13 @@ function getUserProfile (authToken, request, userUuid) {
 }
 
 function getUpgradeOptions (request, userUuid, userProfileResult) {
-	return membershipApi.fetchUpgradeOptions({userId: userUuid, offerId: request.offerId, countryCode: userProfileResult.user.homeAddress.country, apiEnv: request.apiEnv});
+	return membershipApi.fetchUpgradeOptions({userId: userUuid, offerId: request.offerId, countryCode: userProfileResult.user.homeAddress.country, apiEnv: request.apiEnv})
+		.then(result => result)
+		.catch(error => error);
```
Comment (geek-caroline) [inline server/controllers/subscription-transitions/get-upgrade-options-insecure.js]: I don't quite understand how this error is handled, is it automatically handled by the `and.on('error... ` bit in line 18?

File: server/controllers/subscription-transitions/get-upgrade-options-insecure.js [hunk 3]
```diff
@@ -53,7 +58,13 @@ function getUserProfile (authToken, request, userUuid) {
...
+}
+
+function offerOrUserHasNotBeenFound (apiResult) {
+	return !apiResult || !apiResult.upgradeOptions;
 }
```
Comment (andygout) [inline server/controllers/subscription-transitions/get-upgrade-options-insecure.js]: Arrow functions FTW: `const offerOrUserHasNotBeenFound = apiResult => !apiResult || !apiResult.upgradeOptions;`.

File: server/controllers/subscription-transitions/get-upgrade-options-insecure.js [hunk 4]
```diff
@@ -67,3 +78,7 @@ function ApiResult (apiResult) {
 		upgradeType
 	};
 }
+
+function getApiErrorResponse (apiResult) {
+	return ApiResponse.of(404, apiResult.data);
+}
```
Comment (andygout) [inline server/controllers/subscription-transitions/get-upgrade-options-insecure.js]: Do as arrow function as well?

File: server/controllers/subscription-transitions/get-upgrade-options-insecure.js [hunk 5]
```diff
@@ -53,7 +58,13 @@ function getUserProfile (authToken, request, userUuid) {
 }
 
 function getUpgradeOptions (request, userUuid, userProfileResult) {
-	return membershipApi.fetchUpgradeOptions({userId: userUuid, offerId: request.offerId, countryCode: userProfileResult.user.homeAddress.country, apiEnv: request.apiEnv});
+	return membershipApi.fetchUpgradeOptions({userId: userUuid, offerId: request.offerId, countryCode: userProfileResult.user.homeAddress.country, apiEnv: request.apiEnv})
+		.then(result => result)
+		.catch(error => error);
+}
```
Comment (andygout) [inline server/controllers/subscription-transitions/get-upgrade-options-insecure.js]: Also as arrow function?

**Comment by tavvy** [comment]
Unfortunately due to the nature of routine and how it ~abuses~ uses `this` binding and scope, using arrow functions would cause problems

**Comment by tavvy** [comment]
@andygout @geek-caroline I have updated this PR a bit.
>Add ability to throw "handled" errors 
In a scenario where we get a failure response from an api.
We can return a 500 but NOT report the error to sentry.

**Pull request merged**